data/src/main/java/org/apache/iceberg/data/parquet/GenericParquetReaders.java

```diff
@@ -396,7 +409,12 @@ private FixedReader(ColumnDescriptor desc) {
     }
   }
 
-  static class RecordReader extends StructReader<Record, Record> {
+  public interface StructReaderFactory<T> {
```

Comment:
Similar to the write path, I think it would be nice to refactor this to avoid exposing new public methods and interfaces.